spec/02-integration/04-admin_api/06-certificates_routes_spec.lua

```diff
@@ -156,31 +176,310 @@ describe("Admin API: #" .. kong_config.database, function()
+        assert.is_string(json.key)
+        assert.same({"baz.com"}, json.snis)
+
+        -- make sure we added an sni
+        res = assert(client:send {
+          method  = "GET",
+          path    = "/snis",
+        })
+
+        body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.equal(3, #json.data)
+        assert.equal(3, json.total)
+
+        -- make sure we added our certificate
+        res = assert(client:send {
           method = "GET",
           path = "/certificates",
         })
 
         local body = assert.res_status(200, res)
         local json = cjson.decode(body)
-        assert.equal(1, json.total)
-        assert.equal(1, #json.data)
-        assert.is_string(json.data[1].cert)
-        assert.is_string(json.data[1].key)
-        assert.same({ "foo.com", "bar.com" }, json.data[1].snis)
+        assert.equal(3, #json.data)
+        assert.equal(3, json.total)
+      end)
+
+      it("returns 404 for a random non-existing id", function()
+
+        local res = assert(client:send {
+          method  = "PUT",
+          path    = "/certificates",
+          body    = {
+            id = utils.uuid(),
+            cert  = ssl_fixtures.cert,
+            key   = ssl_fixtures.key,
+            snis  = "baz.com",
+          },
+          headers = { ["Content-Type"] = "application/x-www-form-urlencoded" },
+        })
+
+        assert.res_status(404, res)
+
+        -- make sure we did not add any sni
+        res = assert(client:send {
+          method  = "GET",
+          path    = "/snis",
+        })
+
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.equal(2, #json.data)
+        assert.equal(2, json.total)
+
+        -- make sure we did not add any certificate
+        res = assert(client:send {
+          method = "GET",
+          path = "/certificates",
+        })
+
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        assert.equal(2, json.total)
+        assert.equal(2, #json.data)
+      end)
+
+      it("updates snis associated with a certificate", function()
+
+        local res = assert(client:send {
+          method  = "PUT",
+          path    = "/certificates",
+          body    = {
+            id = cert_foo.id,
+            snis  = "baz.com",
+          },
+          headers = { ["Content-Type"] = "application/x-www-form-urlencoded" },
+        })
+
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+
+        assert.same({"baz.com" }, json.snis)
+
+        -- make sure number of snis don't change
+        -- since we delete foo.com and added baz.com
+        res = assert(client:send {
+          method  = "GET",
+          path    = "/snis",
+        })
+
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        assert.equal(2, #json.data)
+        assert.equal(2, json.total)
```

Comment:
Could be worth checking if our `snis` update was effective in the DB itself and checking the response body from `GET` here maybe?